client/src/components/NavLinks/NavLinks.js

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import { NavLink } from 'react-router-dom';
+import HomeIcon from '../Icons/HomeIcon/HomeIcon';
+import ChatIcon from '../Icons/ChatIcon/ChatIcon';
+import ExploreIcon from '../Icons/ExploreIcon/ExploreIcon';
+import styles from './NavLinks.module.scss';
+import HeartIcon from '../Icons/HeatIcon/HeartIcon';
+import ProfilePic from '../ProfilePic/ProfilePic';
+
+const NavLinks = () => {
+	const [
+		homeIcon,
+		setHomeIcon
+	] = useState(true);
+
+	const [
+		chatIcon,
+		setChatIcon
+	] = useState(true);
+
+	const [
+		exploreIcon,
+		setExploreIcon
+	] = useState(true);
+
+	return (
+		<div className={styles.NavLinks}>
+			<NavLink
+				exact
+				to='/'
+				isActive={match => {
+					if (!match) {
+						setHomeIcon(false);
+					} else {
+						setHomeIcon(true);
+					}
+				}}
```

Comment:
You could extract all of these into a separate function that returns another function that does this like so:
```js

const isActive = (setter) => (match) => setter(!!match)

return (
	<div className={styles.NavLinks}>
			<NavLink
				exact
				to='/'
				isActive={isActive(setHomeIcon)}
			>
```

This is a bit of a fun exercise in functional programming thinking.